developer/src/kmcmplib/src/compfile.h

```diff
@@ -71,85 +72,95 @@
-	KMX_BOOL fIsOption;
-	KMX_BOOL fIsDebug;
-	KMX_BOOL fIsCall;
-	int line;
-	};
-
-typedef FILE_STORE * PFILE_STORE;
-
+  KMX_DWORD dwSystemID;
+  KMX_WCHAR szName[SZMAX_STORENAME];	// the name of the store
+  PKMX_WCHAR dpString;	    				// from start of store structure
+  //FileStoreType fstType;
+  KMX_BOOL fIsStore;
+  KMX_BOOL fIsReserved;
+  KMX_BOOL fIsOption;
+  KMX_BOOL fIsDebug;
+  KMX_BOOL fIsCall;
+  int line;
+  };
+
+typedef FILE_STORE *PFILE_STORE;
 
 struct FILE_KEY {
-	KMX_WCHAR   Key;            // WCHAR for consistency; only a byte used however
-	KMX_WORD    LineStoreIndex;
-	KMX_DWORD   Line;
-	KMX_DWORD   ShiftFlags;
-	PKMX_WCHAR  dpOutput;		// from start of key structure
-	PKMX_WCHAR  dpContext;		// from start of key structure
-	};
+  KMX_WCHAR   Key;            // WCHAR for consistency; only a byte used however
+  KMX_WORD    LineStoreIndex;
+  KMX_DWORD   Line;
+  KMX_DWORD   ShiftFlags;
+  PKMX_WCHAR  dpOutput;		// from start of key structure
+  PKMX_WCHAR  dpContext;		// from start of key structure
+  };
 typedef FILE_KEY *PFILE_KEY;
 
-
 struct FILE_GROUP {
-	KMX_WCHAR		szName[SZMAX_GROUPNAME];
-	PFILE_KEY	dpKeyArray;         // address of first item in key array, from start of group structure
-	PKMX_WCHAR      dpMatch;             // from start of group structure
-	PKMX_WCHAR      dpNoMatch;           // from start of group structure
-	KMX_DWORD cxKeyArray;               // in array items
-	KMX_BOOL  fUsingKeys;               // group(xx) [using keys] <-- specified or not
-	KMX_BOOL  fReadOnly;                // group(xx) [readonly] <-- specified or not
-    KMX_DWORD Line;
-	};
+  KMX_WCHAR		szName[SZMAX_GROUPNAME];
+  PFILE_KEY	dpKeyArray;         // address of first item in key array, from start of group structure
+  PKMX_WCHAR      dpMatch;             // from start of group structure
+  PKMX_WCHAR      dpNoMatch;           // from start of group structure
+  KMX_DWORD cxKeyArray;               // in array items
+  KMX_BOOL  fUsingKeys;               // group(xx) [using keys] <-- specified or not
+  KMX_BOOL  fReadOnly;                // group(xx) [readonly] <-- specified or not
+  KMX_DWORD Line;
+};
 typedef FILE_GROUP *PFILE_GROUP;
 
 struct FILE_DEADKEY
 {
-	KMX_WCHAR szName[SZMAX_DEADKEYNAME];
+  KMX_WCHAR szName[SZMAX_DEADKEYNAME];
 };
+
 typedef FILE_DEADKEY *PFILE_DEADKEY;
 
 struct FILE_VKDICTIONARY
 {
-	KMX_WCHAR szName[SZMAX_VKDICTIONARYNAME];
+  KMX_WCHAR szName[SZMAX_VKDICTIONARYNAME];
 };
 typedef FILE_VKDICTIONARY *PFILE_VKDICTIONARY;
 
+/**
+ * Extra metadata for API consumers
+ */
+struct FILE_KEYBOARD_EXTRA {
+  std::u16string kvksFilename;	// original TSS_VISUALKEYBOARD value
+};
+
+typedef struct FILE_KEYBOARD_EXTRA* PFILE_KEYBOARD_EXTRA;
+
 struct FILE_KEYBOARD {
-	KMX_DWORD KeyboardID;			// as stored in HKEY_LOCAL_MACHINE//system//currentcontrolset//control//keyboard layouts
+  KMX_DWORD KeyboardID;			// as stored in HKEY_LOCAL_MACHINE//system//currentcontrolset//control//keyboard layouts
```

Comment:
```suggestion
  KMX_DWORD KeyboardID;			// as stored in HKEY_LOCAL_MACHINE/system/currentcontrolset/control/keyboard layouts
```

Reply:
actually that's a really old comment, marking now as 'deprecated'